src/components/navBar/navBar.js

```diff
@@ -134,47 +178,183 @@ function MdNavBar($mdAria) {
-  this._$mdConstant = $mdConstant;
+    /** @type {Function} */
+    this.canPageForward = canPageForward;
 
-  // Data-bound variables.
-  /** @type {string} */
-  this.mdSelectedNavItem;
+    /** @type {Number} */
+    this.left = 0;
 
-  /** @type {string} */
-  this.navBarAriaLabel;
 
-  // State variables.
+    /**
+     * Gathers references to all of the DOM elements used by this controller.
+     * @returns {Object}
+     */
+    function getElements() {
+        var elements = {};
+        var node = $element[0];
+        elements.parent = node.parentElement;
+        elements.canvas = node.querySelector('.md-nav-bar');
+        elements.tabs = elements.canvas.querySelectorAll('.md-nav-item');
 
-  /** @type {?angular.JQLite} */
-  this._navBarEl = $element[0];
+        return elements;
+    }
 
-  /** @type {?angular.JQLite} */
-  this._inkbar;
+    /**
+     * Updates whether or not pagination should be displayed
+     * and the pagination styles defined by the element.
+     */
+    function updatePagination() {
+        self.shouldPaginate = shouldPaginate();
+
+        // Setting the style.
+        var rootElement = $element[0];
+        var prevButton = angular.element(rootElement.querySelector('md-prev-button'));
+        var nextButton = angular.element(rootElement.querySelector('md-next-button'));
+        prevButton.css({
+            'background-image': 'linear-gradient(to right, ' + $element.css('background-color') + ', rgba(0,0,0,0.0))'
+        });
+        nextButton.css({
+            'background-image': 'linear-gradient(to left, ' + $element.css('background-color') + ', rgba(0,0,0,0.0))'
+        });
 
-  var self = this;
-  // need to wait for transcluded content to be available
-  var deregisterTabWatch = this._$scope.$watch(function() {
-    return self._navBarEl.querySelectorAll('._md-nav-button').length;
-  },
-  function(newLength) {
-    if (newLength > 0) {
-      self._initTabs();
-      deregisterTabWatch();
+        if (!self.shouldPaginate) {
+            var canvas = angular.element(getElements().canvas);
+            self.left = 0;
+            canvas.css({
+                'left': self.left + 'px'
+            });
+        }
+    }
+
+    /**
+     * Calculate the real width size of the mdNavBar using each tab of it.
+     * @returns {number}
+     */
+    function getNavWidth() {
+        var tabs = getElements().tabs
+        var contWidth = 0;
+        for (var i = 0; i < tabs.length; i++) {
+            var tab = tabs[i];
+            contWidth = contWidth + tab.offsetWidth;
+        }
+        return contWidth;
+    }
+
+    /**
+     * Determines if pagination is necessary to display the tabs within the available space.
+     * @returns {boolean}
+     */
+    function shouldPaginate() {
+        var elements = getElements();
+        var navWidth = getNavWidth();
+        var parentWidth = elements.parent != null && elements.parent.clientWidth != null ? elements.parent.clientWidth : 0;
```

Comment:
This is a confusing line. Consider simplifying by using a real if statement and using !!elements.parent for the nullability checks.